app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -122,6 +122,10 @@ class UserViewModel(
 
     Log.d("UserViewModel", "Updating user: $user")
     // First update the user object in the view model.
+    if (currentUser.value == user) {
+      Log.d("UserViewModel", "No difference between the current user and the user to update")
```

Comment:
The wallet is not immutable.
If the user number of coin changes and only that, I believe this condition will evaluate to true and the update be skipped. 

I think we need to redefine the equal method of the user class if we want to check for equality

Reply:
I think you can  remove the optimi. 
The update user is a very small fraction of all our API calls

Reply:
Done :(